install_debs_into_image.sh

```diff
@@ -0,0 +1,189 @@
+else
+	echo "Only a reduced application set will be built into the given image."
+fi
+
+IMAGEDIR=`mktemp -d -p /tmp img.XXXXXXXX`
+BAKERYDIR=$(dirname "$0")
+LOOPDEVICE=$(losetup -f)
+CONFIGTXT="$IMAGEDIR/boot/config.txt"
+
+cleanup_umount() {
+	if [ -e "$IMAGEDIR" ] ; then
+		lsof -t "$IMAGEDIR" | xargs --no-run-if-empty kill
+	fi
+	if [ -e "$IMAGEDIR/usr/bin/qemu-arm-static" ] ; then
+		rm -f "$IMAGEDIR/usr/bin/qemu-arm-static"
+	fi
+	if mountpoint -q "$IMAGEDIR/tmp/debs-to-install" ; then
+		umount "$IMAGEDIR/tmp/debs-to-install"
+	fi
+	if [ -e "$IMAGEDIR/tmp/debs-to-install" ] ; then
+		rmdir "$IMAGEDIR/tmp/debs-to-install"
+	fi
+	if mountpoint -q "$IMAGEDIR/boot" ; then
+		umount "$IMAGEDIR/boot"
+	fi
+	if mountpoint -q "$IMAGEDIR" ; then
+		umount "$IMAGEDIR"
+	fi
+	if [ -d "$IMAGEDIR" ] ; then
+		rmdir "$IMAGEDIR"
+	fi
+}
+
+cleanup_losetup() {
+	if [ -e "$LOOPDEVICE"p1 ] ; then
+		delpart "$LOOPDEVICE" 1
+	fi
+	if [ -e "$LOOPDEVICE"p2 ] ; then
+		delpart "$LOOPDEVICE" 2
+	fi
+	if losetup "$LOOPDEVICE" 2>/dev/null ; then
+		losetup -d "$LOOPDEVICE"
+	fi
+}
+
+cleanup() {
+	cleanup_umount
+	cleanup_losetup
+}
+
+trap cleanup ERR SIGINT
+imgsize=$($PARTED "$1" unit b print | grep -e "\.img" | awk -F ":" '{gsub(/^[ \t]+|[B \t]+$/,"",$2); print $2}')
+[ "x$imgsize" = "x" ] && echo 1>&1 "Error: Image size not found" && exit 1
+secsize=$($PARTED "$1" unit b print | grep -e "Sector size" | awk -F "/" '{gsub(/^[ \t]+|[B \t]+$/,"",$3); print $3}')
+[ "x$secsize" = "x" ] && echo 1>&1 "Error: Sector size not found" && exit 1
+
+# The smallest size of CM3-eMMC is 4 GB  , the available disksize for a system is 3909091328 bytes
+# An image like raspios-lite just have 2 GB , so we need to resize rootfs first before
+# we start processing the image. Otherwise build will fail with "no space left on device"
+if [ $imgsize -lt 3900000000 ] ; then
+	disksize=3909091328
+	bcount=$(echo "($disksize-$imgsize)/$secsize" | bc )
+	dd if=/dev/zero count=$bcount bs=$secsize >> "$1"
+	$PARTED "$1" resizepart 2 "$((disksize-1))"B
+	losetup "$LOOPDEVICE" "$1"
+	partprobe "$LOOPDEVICE"
+	resize2fs "$LOOPDEVICE"p2
+	e2fsck -p -f "$LOOPDEVICE"p2
+	sync
+	losetup -D
+fi
+
+# mount ext4 + FAT filesystems
+losetup "$LOOPDEVICE" "$1"
+partprobe "$LOOPDEVICE"
+mount "$LOOPDEVICE"p2 "$IMAGEDIR"
+mount "$LOOPDEVICE"p1 "$IMAGEDIR/boot"
+
+# see https://wiki.debian.org/QemuUserEmulation
+if [ -e /usr/bin/qemu-arm-static ] ; then
+	cp /usr/bin/qemu-arm-static "$IMAGEDIR/usr/bin"
+fi
+
+# Move ld.so.preload until installation is finished. Otherwise we get errors
+# from ld.so:
+#   ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so'
+#   from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
+mv "$IMAGEDIR/etc/ld.so.preload" "$IMAGEDIR/etc/ld.so.preload.bak"
+
+# customize settings
+echo `basename "$1"` > "$IMAGEDIR/etc/revpi/image-release"
+
+chroot "$IMAGEDIR" apt-get update
+chroot "$IMAGEDIR" apt-get -y install revpipyload
```

Comment:
@nbuchwitz  should the installation of revpipyload be replaced with package list from debs-to-download? or just remove it and apt-get update to eliminate the dependency of internet connectivity ?  or just keep it as it now?

Reply:
Yes, please use debs-to-download